pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -188,7 +188,8 @@ func (e *endpointImpl) stop() {
 
 // dial establishes the gRPC communication with the registered device plugin.
 func dial(unixSocketPath string) (pluginapi.DevicePluginClient, *grpc.ClientConn, error) {
```

Comment:
How about moving this function to device_plugin_stub.go add a timeout parameter and return only a *grpc.ClientConn ?
Add a comment about why we are using WithBlock()

Reply:
Why is the rationale behind the suggestion of moving this function to stub.go? Since it is being invoked from within this file, https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/deviceplugin/endpoint.go#L58,
I think this file is the right place for this function.

IMO, `WithBlock` is self explanatory. As far as scope of this PR goes, i think enough details have already been added in PR description.

Reply:
> Why is the rationale behind the suggestion of moving this function to stub.go? Since it is being invoked from within this file.

Rational is to call it on the stub. Wether we move it or not is not really the important thing here.

> IMO, WithBlock is self explanatory. As far as scope of this PR goes, i think enough details have already been added in PR description.

When people read the code, they don't read the PRs. Adding a comment explaining why this is required and what this is fixing helps understands why these options are here.